core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -2593,6 +2593,31 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
 
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6592">[CALCITE-6592]
+   * RelMdPredicates pull up wrong predicate from UNION when it's input include NULL VALUE</a>. */
+  @Test void testPullUpPredicatesFromUnionWithValues1() {
+    final String sql = "select cast(null as integer) as a\n"
+        + "union all\n"
+        + "select 5 as a";
+    final Union rel = (Union) sql(sql).toRel();
+    final RelMetadataQuery mq = rel.getCluster().getMetadataQuery();
+    RelOptPredicateList inputSet = mq.getPulledUpPredicates(rel);
+    ImmutableList<RexNode> pulledUpPredicates = inputSet.pulledUpPredicates;
+    assertThat(pulledUpPredicates, sortsAs("[OR(null, =($0, 5))]"));
```

Comment:
The result does not look 100% accurate. Shouldn't this rather be `OR(IS NULL($0), =($0, 5))`. What happens if there were more columns involved?

Reply:
@zabetak Hi, testPullUpPredicatesFromUnionWithValues3 added for more columns. Maybe we should extract nothing when we can't get a specific value like OR(IS NULL($0), =($0, 5)). WDYT?